core-tests/jvm/src/test/scala/zio/IOSpec.scala

```diff
@@ -305,6 +307,21 @@ class IOSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntim
     )
   }
 
+  def testCached = {
+    def incrementAndGet(ref: Ref[Int]): UIO[Int] = ref.modify(n => (n + 1, n + 1))
+    unsafeRun {
+      for {
+        ref   <- Ref.make(0)
+        cache <- incrementAndGet(ref).cached(100.milliseconds)
+        a     <- cache
+        b     <- cache
+        _     <- clock.sleep(100.milliseconds)
```

Comment:
This will be flaky due to non-determinism. We could use with `MockClock` or add `flaky` combinator to test (it's defined in `RTSSpec`).

Reply:
Yes, trying to test this is actually what led me to identify the issues with the `MockClock`. I used the `flaky` combinator for now. If we get the `MockClock` improvements in first we can switch it to use that or otherwise we can do it subsequently.